project/cmake/modules/FindFFMPEG.cmake

```diff
@@ -1,18 +1,140 @@
+                                libavutil/avutil.h libswscale/swscale.h libpostproc/postprocess.h
+            PATH_SUFFIXES ffmpeg
+            PATHS ${PC_FFMPEG_INCLUDE_DIRS}
+            NO_DEFAULT_PATH)
+  find_path(FFMPEG_INCLUDE_DIRS libavcodec/avcodec.h libavfilter/avfilter.h libavformat/avformat.h
+                                libavutil/avutil.h libswscale/swscale.h libpostproc/postprocess.h)
+
+  find_library(FFMPEG_LIBAVCODEC
+               NAMES avcodec libavcodec
+               PATH_SUFFIXES ffmpeg/libavcodec
+               PATHS ${PC_FFMPEG_libavcodec_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBAVCODEC NAMES avcodec libavcodec PATH_SUFFIXES ffmpeg/libavcodec)
+
+  find_library(FFMPEG_LIBAVFILTER
+               NAMES avfilter libavfilter
+               PATH_SUFFIXES ffmpeg/libavfilter
+               PATHS ${PC_FFMPEG_libavfilter_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBAVFILTER NAMES avfilter libavfilter PATH_SUFFIXES ffmpeg/libavfilter)
+
+  find_library(FFMPEG_LIBAVFORMAT
+               NAMES avformat libavformat
+               PATH_SUFFIXES ffmpeg/libavformat
+               PATHS ${PC_FFMPEG_libavformat_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBAVFORMAT NAMES avformat libavformat PATH_SUFFIXES ffmpeg/libavformat)
+
+  find_library(FFMPEG_LIBAVUTIL
+               NAMES avutil libavutil
+               PATH_SUFFIXES ffmpeg/libavutil
+               PATHS ${PC_FFMPEG_libavutil_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBAVUTIL NAMES avutil libavutil PATH_SUFFIXES ffmpeg/libavutil)
+
+  find_library(FFMPEG_LIBSWSCALE
+               NAMES swscale libswscale
+               PATH_SUFFIXES ffmpeg/libswscale
+               PATHS ${PC_FFMPEG_libswscale_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBSWSCALE NAMES swscale libswscale PATH_SUFFIXES ffmpeg/libswscale)
+
+  find_library(FFMPEG_LIBSWRESAMPLE
+               NAMES swresample libswresample
+               PATH_SUFFIXES ffmpeg/libswresample
+               PATHS ${PC_FFMPEG_libswresample_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBSWRESAMPLE NAMES NAMES swresample libswresample PATH_SUFFIXES ffmpeg/libswresample)
+
+  find_library(FFMPEG_LIBPOSTPROC
+               NAMES postproc libpostproc
+               PATH_SUFFIXES ffmpeg/libpostproc
+               PATHS ${PC_FFMPEG_libpostproc_LIBDIR}
+               NO_DEFAULT_PATH)
+  find_library(FFMPEG_LIBPOSTPROC NAMES postproc libpostproc PATH_SUFFIXES ffmpeg/libpostproc)
+
+  if(PC_FFMPEG_FOUND
+     AND PC_FFMPEG_libavcodec_VERSION
+     AND PC_FFMPEG_libavfilter_VERSION
+     AND PC_FFMPEG_libavformat_VERSION
+     AND PC_FFMPEG_libavutil_VERSION
+     AND PC_FFMPEG_libswscale_VERSION
+     AND PC_FFMPEG_libswresample_VERSION
+     AND PC_FFMPEG_libpostproc_VERSION)
+    set(FFMPEG_VERSION 3.1)
```

Comment:
I would set a variable at the top where you define FFMPEG_PKGS, that way if we bump the version, the version variables are in once place.